collect_app/src/main/java/org/odk/collect/android/formlists/blankformlist/BlankFormListViewModel.kt

```diff
@@ -0,0 +1,240 @@
+import org.odk.collect.android.preferences.utilities.FormUpdateMode
+import org.odk.collect.android.preferences.utilities.SettingsUtils
+import org.odk.collect.android.utilities.ChangeLockProvider
+import org.odk.collect.android.utilities.FormsDirDiskFormsSynchronizer
+import org.odk.collect.androidshared.data.Consumable
+import org.odk.collect.androidshared.livedata.LiveDataUtils
+import org.odk.collect.androidshared.livedata.MutableNonNullLiveData
+import org.odk.collect.async.Scheduler
+import org.odk.collect.forms.FormSourceException
+import org.odk.collect.forms.FormSourceException.AuthRequired
+import org.odk.collect.forms.FormsRepository
+import org.odk.collect.settings.keys.ProjectKeys
+import org.odk.collect.shared.settings.Settings
+import org.odk.collect.shared.strings.Md5.getMd5Hash
+import java.io.ByteArrayInputStream
+
+class BlankFormListViewModel(
+    private val formsRepository: FormsRepository,
+    private val application: Application,
+    private val syncRepository: SyncStatusAppState,
+    private val formsUpdater: FormsUpdater,
+    private val scheduler: Scheduler,
+    private val generalSettings: Settings,
+    private val analytics: Analytics,
+    private val changeLockProvider: ChangeLockProvider,
+    private val formsDirDiskFormsSynchronizer: FormsDirDiskFormsSynchronizer,
+    private val projectId: String
+) : ViewModel() {
+
+    private val _allForms: MutableNonNullLiveData<List<BlankFormListItem>> = MutableNonNullLiveData(emptyList())
+    private val _formsToDisplay: MutableLiveData<List<BlankFormListItem>> = MutableLiveData()
+    val formsToDisplay: LiveData<List<BlankFormListItem>> = _formsToDisplay
+
+    private val _syncResult: MutableLiveData<Consumable<String>> = MutableLiveData()
+    val syncResult: LiveData<Consumable<String>> = _syncResult
+
+    private val isFormLoadingRunning = MutableNonNullLiveData(false)
+    private val isSyncingWithStorageRunning = MutableNonNullLiveData(false)
+    private val isSyncingWithServerRunning = MutableNonNullLiveData(false)
+
+    val backgroundTasksObserver: LiveData<Triple<Boolean, Boolean, Boolean>> = LiveDataUtils.zip3(
+        isFormLoadingRunning,
+        isSyncingWithStorageRunning,
+        isSyncingWithServerRunning,
+    )
+
+    var sortingOrder: Int = generalSettings.getInt("formChooserListSortingOrder")
+        get() { return generalSettings.getInt("formChooserListSortingOrder") }
+
+        set(value) {
+            field = value
+            generalSettings.save("formChooserListSortingOrder", value)
+            sortAndFilter()
+        }
+
+    var filterText: String = ""
+        set(value) {
+            field = value
+            sortAndFilter()
+        }
+
+    private val shouldHideOldFormVersions: Boolean
+        get() {
+            return generalSettings.getBoolean(ProjectKeys.KEY_HIDE_OLD_FORM_VERSIONS)
+        }
+
+    init {
+        loadFromDatabase()
```

Comment:
It's not required but I thought that maybe we should load data from database first in case when syncing would take a lot of time. I'm pretty sure the old approach worked like that. Do you think we should wait for syncing?

Reply:
> that maybe we should load data from database first in case when syncing would take a lot of time

Hmmm yeah I think I agree with you. Can we add a test that checks that loading happens first, and then we sync (and reload)? That should be possible using `FakeScheduler`.

Reply:
Done.